client/src/views/files/FileOperationMenu.vue

```diff
@@ -60,14 +60,19 @@
     <ion-list class="upload-menu-list">
       <template v-if="currentTab === Tabs.InProgress">
         <component
-          v-for="item in inProgressItems"
+          v-for="item in inProgressItems.slice(0, MAX_SIMULTANEOUS_OPERATIONS)"
```

Comment:
I'm not sure this list is always ordered. Instead, use another list for items that have the state `OperationProgress` (and maybe rename the `inProgressItems` list.